website/scripts/ros/ros_traffic_signal.js

```diff
@@ -1,67 +1,75 @@
-                        break;
-                    case SIGNAL_RED_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('red',element.remaining_time));
-                        break;
-                    case SIGNAL_YELLOW_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('yellow',element.remaining_time));
-                        break;
-                    case SIGNAL_FLASHING_GREEN_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('flash_green',element.remaining_time));
-                        break;
-                    case SIGNAL_FLASHING_YELLOW_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('flash_yellow',element.remaining_time));
-                        break;
-                    case SIGNAL_FLASHING_RED_STATE:
-                        $('.traffic-signal-col').append(updateTrafficSignal('flash_red',element.remaining_time));
-                        break;
-                    default:
-                        $('.traffic-signal-col').html('');
-                        $('.traffic-signal-col').append(updateTrafficSignal('',''));
-                        console.error("Traffic signal state is invalid");
-                        break;
-                } 
-                //set back to black after 5 seconds.
-                if(!isTimeerSet)
-                {
-                    isTimeerSet=true;
-                    setTimeout(function(){
-                        $('.traffic-signal-col').html('');
-                        $('.traffic-signal-col').append(updateTrafficSignal('',''));
-                        isTimeerSet=false;
-                        signalState = null;
-                    }, 5000);
-                }
-           }      
-        });
-            
+         try{
+            if(Array.isArray(message.intersections.intersection_state_list)){
+                message.intersections.intersection_state_list.forEach(element=>{
+                    element.states.movement_list.forEach(inner_ele=>{
+                        inner_ele.state_time_speed.movement_event_list.forEach(event_ele=>{
+                            let signal_state = event_ele.event_state.movement_phase_state;
+                            let remaining_time = event_ele.timing.min_end_time;
+                            //Prevent repeating the same state
+                            if(signalState !=signal_state)
+                            {
+                                signalState = signal_state;
+                                    switch(signalState)
+                                    {
+                                        case TRAFFIC_SIGNAL_PHASE_STATE.protected_movement_allowed:
+                                            $('.traffic-signal-col').append(updateTrafficSignal('green',remaining_time));
```

Comment:
Is append() really the right operation here? Wouldn't some sort of assignment operation be more appropriate? Wouldn't this case and output that adds a new entry every time the phase state changes?